metadata-ingestion/src/datahub/ingestion/source/mode.py

```diff
@@ -456,9 +463,22 @@ def construct_dashboard(
         # Datasets
         datasets = []
         for imported_dataset_name in report_info.get("imported_datasets", {}):
-            mode_dataset = self._get_request_json(
-                f"{self.workspace_uri}/reports/{imported_dataset_name.get('token')}"
-            )
+            try:
+                mode_dataset = self._get_request_json(
+                    f"{self.workspace_uri}/reports/{imported_dataset_name.get('token')}"
+                )
+            except HTTPError as http_error:
+                status_code = http_error.response.status_code
+                if status_code == 404:
+                    self.report.report_warning(
+                        title="Missing Dataset",
+                        message="Can not get dataset by token",
+                        context=f"Report token: {report_token}",
```

Comment:
the error messages could be improved - say what this would impact in a way a nontechnical user would understand